src/pages/wallet-connect/wallet-connect.ts

```diff
@@ -0,0 +1,434 @@
+          case 'eth_signTransaction':
+            transaction = this.displayRequest.params[0];
+            addressRequested = transaction.from;
+            if (this.address.toLowerCase() === addressRequested.toLowerCase()) {
+              result = await this.walletProvider.publishAndSign(
+                this.wallet,
+                transaction
+              );
+            } else {
+              errorMsg = 'Address requested does not match active account';
+            }
+            break;
+          default:
+            break;
+        }
+
+        if (result) {
+          this.walletConnector.approveRequest({
+            id: this.displayRequest.id,
+            result
+          });
+        } else {
+          let message = 'JSON RPC method not supported';
+          if (!this.getWallet()) {
+            message = 'No Active Account';
+          }
+          this.walletConnector.rejectRequest({
+            id: this.displayRequest.id,
+            error: { message }
+          });
+        }
+      }
+    } catch (error) {
+      this.logger.error(error);
+      if (this.walletConnector) {
+        this.walletConnector.rejectRequest({
+          id: this.displayRequest.id,
+          error: { message: errorMsg || 'Failed or Rejected Request' }
+        });
+      }
+    }
+
+    this.closeRequest();
+  }
+
+  public updateWallet(index: number, chainId: number) {
+    this.activeIndex = index;
+    this.activeChainId = chainId;
+    const rpcUrl = this.getChainData(chainId).rpc_url;
+    this.wallet = this.generateWallet(index);
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    this.wallet.connect(provider);
+    return this.wallet;
+  }
+
+  public getChainData(chainId: number): IProviderData {
+    const chainData = supportedProviders.filter(
+      (chain: any) => chain.chain_id === chainId
+    )[0];
+
+    if (!chainData) {
+      throw new Error('ChainId missing or not supported');
+    }
+
+    const API_KEY = 'f73d326fec924c34abc4d16233920356';
+
+    if (
+      chainData.rpc_url.includes('infura.io') &&
+      chainData.rpc_url.includes('%API_KEY%') &&
+      API_KEY
+    ) {
+      const rpcUrl = chainData.rpc_url.replace('%API_KEY%', API_KEY);
+
+      return {
+        ...chainData,
+        rpc_url: rpcUrl
+      };
+    }
+
+    return chainData;
+  }
+
+  public getWallet(index?: number, chainId?: number): ethers.Wallet {
+    let _wallet = this.wallet;
+    if (
+      !_wallet ||
+      this.activeIndex === index ||
+      this.activeChainId === chainId
+    ) {
+      _wallet = this.updateWallet(index, chainId);
+    }
+    return _wallet;
+  }
+
+  public generateWallet(account: number) {
+    /* TODO wrap wallet private key or mnemonic with ethers wallet.
+     * Move this function to key.ts in BWC and use that function here.
+     */
+    const newWallet = ethers.Wallet.fromMnemonic(
+      'input your mnemonic here for testing',
+      `m/44'/60'/${account}'/0/0`
+    );
```

Comment:
TODO:

Need to move this function to BWC in `key.ts` then expose the function to Copay where we can use that function here. 

To test add an ethereum livenet wallet mnemonic.